tests/lib.rs

```diff
@@ -562,6 +562,34 @@ fn test_pager_flag_enable() {
         .success();
 }
 
+#[test]
+fn test_multiple_platform_command_search() {
+    let testenv = TestEnv::new();
+    testenv.add_os_entry("linux", "linux-only", "this command only exists for linux");
+
+    testenv
+        .command()
+        .args(["--platform", "macos", "--platform", "linux", "linux-only"])
```

Comment:
Do we want to guarantee behavior if the page occurs for multiple platforms? Maybe try platforms in the order they were specified? In that case, we would also need a test for that